src/WrkLst/DocxMustache/DocxMustache.php

```diff
@@ -146,21 +146,35 @@ protected function SaveOpenXmlObjectToFile($xmlObject, $file, $folder)
     public function ReadTeamplate($dpi)
     {
         $this->Log('Analyze Template');
-        //get the main document out of the docx archive
-        $this->word_doc = $this->ReadOpenXmlFile('word/document.xml', 'file');
-
-        $this->Log('Merge Data into Template');
 
-        $this->word_doc = MustacheRender::render($this->items, $this->word_doc);
+        $this->relevant_files = [];
+        //get every File in docx-Archive
+        $this->zipper
+            ->getRepository()->each([$this, 'retrieveFilesList']);
+        foreach($this->relevant_files as $file) {
+            $this->SubstituteOpenXmlFile($file, $dpi);
+        }
 
-        $this->word_doc = HtmlConversion::convert($this->word_doc);
+        $this->zipper->close();
+    }
 
-        $this->ImageReplacer($dpi);
+    public function retrieveFilesList($file, $stats) {
+        $this->exctractOpenXmlFile($file);
+        if(substr($file,-3) === 'xml' && substr($file,0,4) === 'word') {
```

Comment:
I would like to have a more targeted approach for this to save any kind of additional work (as I see the class already performance lacking). Every XML file seems to be a little overkill, as far as I can tell, as some docx files have many meta files that do not need to be touched. 
I would probably generate something like a whitelist. Relevant files might include xml files in the subfolder word: document.xml, endnotes.xml, footer*.xml, footnotes.xml, header*.xml. There might be more files that are relevant, I haven't done the research on the structure, but the whitelist could be expanded in the future accordingly.
Thoughts?

Reply:
You could be right. I will change my code accordingly.